activerecord/test/cases/transactions_test.rb

```diff
@@ -1386,6 +1386,30 @@ def test_sqlite_add_column_in_transaction
         Topic.reset_column_information
       end
     end
+
+    def test_sqlite_default_transaction_mode_is_immediate
+      assert_queries_match(/BEGIN IMMEDIATE TRANSACTION/i, include_schema: false) do
+        Topic.transaction { Topic.lease_connection.materialize_transactions }
+      end
+    end
+
+    def test_sqlite_transaction_mode_can_be_set_to_deferred
+      assert_queries_match(/BEGIN DEFERRED TRANSACTION/i, include_schema: false) do
+        Topic.transaction(mode: :deferred) { Topic.lease_connection.materialize_transactions }
```

Comment:
We shouldn't expose `.transaction(mode:)` It's only needed for transaction fixtures.

Reply:
My thought for exposing it was simply that it is difficult to predict what all applications might need for transactions. `immediate` is *certainly* the correct default, but `deferred` has its place. If an application wants to explicitly begin a `deferred` transaction, do we simply want to document that `joinable: false` produces a deferred transaction?

Reply:
No `joinable` is purely for internal use (transactional fixtures).